include/souper/Extractor/Solver.h

```diff
@@ -41,6 +41,36 @@ class Solver {
                                                    const std::vector<InstMapping> &PCs,
                                                    Inst *LHS, InstContext &IC) = 0;
   virtual std::string getName() = 0;
+  virtual
+  std::error_code negative(const BlockPCs &BPCs,
+                           const std::vector<InstMapping> &PCs,
+                           Inst *LHS, llvm::APInt &Negative,
+                           InstContext &IC) = 0;
+  virtual
+  std::error_code knownBits(const BlockPCs &BPCs,
+                            const std::vector<InstMapping> &PCs,
+                            Inst *LHS, llvm::APInt &Zeros, llvm::APInt &Ones,
+                            InstContext &IC) = 0;
+  virtual
+  std::error_code nonNegative(const BlockPCs &BPCs,
+                            const std::vector<InstMapping> &PCs,
+                            Inst *LHS, llvm::APInt &NonNegative,
+                            InstContext &IC) = 0;
+  virtual
+  std::error_code powerTwo(const BlockPCs &BPCs,
+                            const std::vector<InstMapping> &PCs,
```

Comment:
careful with indentation here and elsewhere -- make sure your browser has no tabs